ruby/private/bundle/bundle.bzl

```diff
@@ -1,123 +1,171 @@
-    environment = {"RUBYOPT": "--disable-gems"}
+BUNDLE_BIN_PATH = "bin"
+BUNDLE_PATH = "lib"
+BUNDLE_BINARY = "bundler/exe/bundler"
+SCRIPT_INSTALL_GEM = "install_bundler.rb"
+SCRIPT_BUILD_FILE_GENERATOR = "create_bundle_build_file.rb"
 
-    result = ctx.execute(args, environment = environment)
-    if result.return_code:
-        message = "Failed to evaluate ruby snippet with {}: {}".format(
-            interpreter,
-            result.stderr,
-        )
-        fail(message)
+# Runs bundler with arbitrary arguments
+#
+# eg: run_bundler(runtime_ctx, [ "lock", " --gemfile", "Gemfile.rails5" ])
+#
+def run_bundler(runtime_ctx, bundler_arguments):
+    # Now we are running bundle install
+    args = [
+        runtime_ctx.interpreter,  # ruby
+        "--enable=gems",  # bundler must run with rubygems enabled
+        "-I",
+        ".",
+        "-I",  # Used to tell Ruby where to load the library scripts
+        BUNDLE_PATH,  # Add vendor/bundle to the list of resolvers
+        BUNDLE_BINARY,  # our binary
+    ] + bundler_arguments
 
-def bundle_install_impl(ctx):
-    ctx.symlink(ctx.attr.gemfile, "Gemfile")
-    ctx.symlink(ctx.attr.gemfile_lock, "Gemfile.lock")
-    ctx.symlink(ctx.attr._create_bundle_build_file, "create_bundle_build_file.rb")
-    ctx.symlink(ctx.attr._install_bundler, "install_bundler.rb")
+    # print("Bundler Command:\n\n", args)
 
-    # TODO(kig) Make Gemspec reference from Gemfile actually work
-    if ctx.attr.gemspec:
-        ctx.symlink(ctx.attr.gemspec, ctx.path(ctx.attr.gemspec).basename)
+    return runtime_ctx.ctx.execute(
+        args,
+        quiet = False,
+        environment = runtime_ctx.environment,
+    )
 
-    ruby = ctx.attr.ruby_interpreter
-    interpreter_path = ctx.path(ruby)
+#
+# Sets local bundler config values by calling
+#
+# $ bundler config --local | --global config-option config-value
+#
+# @config_category can be either 'local' or 'global'
+def set_bundler_config(runtime_ctx, config_category = "local"):
+    # Bundler is deprecating various flags in favor of the configuration.
+    # HOWEVER — for reasons I can't explain, Bazel runs "bundle install" *prior*
+    # to setting these flags. So the flags are then useless until we can force the
+    # order and ensure that Bazel first downloads Bundler, then sets config, then
+    # runs bundle install. Until then, it's a wild west out here.
+    #
+    # Set local configuration options for bundler
+    bundler_config = {
+        "binstubs": BUNDLE_BIN_PATH,
+        "deployment": "'true'",
+        "standalone": "'true'",
+        "frozen": "'true'",
+        "without": "development,test",
+        "path": BUNDLE_PATH,
+        "jobs": "20",
+    }
 
-    install_bundler(
-        ctx,
-        interpreter_path,
-        "install_bundler.rb",
-        "bundler",
-        ctx.attr.version,
-    )
+    for option, value in [(option, value) for option, value in bundler_config.items()]:
```

Comment:
What is the intention with this?

Reply:
As partially described in the comment above, this runs bundler to set the configuration flags in that directory — basically bundler is deprecating command line flags in favor of setting up local config before running bundle install. 

So if today you can:
```
bundle install --deployment --standalone --path=vendor/bundle --jobs=20 --binstubs=bin
```
in the future you will have to:
```
bundle config --local deployment
bundle config --local standalone
bundle config --local path=vendor/bundle
bundle binstubs --path=bin
```
This is what this code does.